web_src/js/features/repo-issue.js

```diff
@@ -437,17 +438,33 @@ export async function handleReply($el) {
 
 export function initRepoPullRequestReview() {
   if (window.location.hash && window.location.hash.startsWith('#issuecomment-')) {
+    // set scrollRestoration to 'manual' when there is a hash in url, so that the scroll position will not be remembered after refreshing
+    if (window.history.scrollRestoration !== 'manual') {
+      window.history.scrollRestoration = 'manual';
+    }
     const commentDiv = $(window.location.hash);
     if (commentDiv) {
       // get the name of the parent id
       const groupID = commentDiv.closest('div[id^="code-comments-"]').attr('id');
       if (groupID && groupID.startsWith('code-comments-')) {
         const id = groupID.slice(14);
+        const ancestorDiffBox = commentDiv.closest('.diff-file-box');
+        // on pages like conversation, there is no diff header
+        const diffHeader = ancestorDiffBox.find('.diff-file-header');
+        // offset is for scrolling
+        let offset = 30;
+        if (diffHeader[0]) {
+          offset += $('.diff-detail-box').outerHeight() + diffHeader.outerHeight();
+        }
         $(`#show-outdated-${id}`).addClass('gt-hidden');
         $(`#code-comments-${id}`).removeClass('gt-hidden');
         $(`#code-preview-${id}`).removeClass('gt-hidden');
         $(`#hide-outdated-${id}`).removeClass('gt-hidden');
-        commentDiv[0].scrollIntoView();
+        // if the comment box is folded, expand it
+        if (ancestorDiffBox.attr('data-folded') && ancestorDiffBox.attr('data-folded') === 'true') {
+          setFileFolding(ancestorDiffBox[0], ancestorDiffBox.find('.fold-file')[0], false);
+        }
+        $('html, body').scrollTop(commentDiv.offset().top - offset);
```

Comment:
Why two selectors? The `scrollTop` should be invoked on a clear target IMO.

ps: I just thought it's a good chance to rewrite the jQuery code to Vanilla JS 😂

Reply:
https://stackoverflow.com/questions/12222485/why-to-use-html-body-for-scrolltop-instead-of-just-html

For the two selectors, I saw this. I think it might be a good idea as we are also using vue, and it sometimes feels strange to have jquery at the same time. But I think I still lack an overall view to the product to comment on this right now..

Reply:
Yep `scrollIntoView` cannot set offset value, added a commit to use `window.scrollto`

Reply:
Sounds good as well.